Assets/Scripts/Items/ItemSpawner.cs

```diff
@@ -0,0 +1,72 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+//[CreateAssetHeader(menu = "Item picker")]
+public class ItemSpawner : MonoBehaviour {
+    [Header("Item Lists")]
+    public List<GameObject> Common_Items;
+    public List<GameObject> Rare_Items,
+                            Legendary_Items;
+
+    [Header("Thresholds")]
+    public int rareThresh = 65;    // numbers above can be rare or legandary
+    public int legendThresh = 95,  // numbers above will be legendary
+               upperThresh = 100;  // marks the highest number that can be generated,
+                                   // could be increased for overall better items
+
+    public bool spawnItem = false;
+    private int weightTest = 0;
+
+    private GameObject spawnedItem;
+
+    void Start() {
+        HandleItem(weightTest);
+    }
+    void Update() {
+        if (Input.GetKeyDown(KeyCode.F4)) {
+            weightTest++;
+            HandleItem(weightTest);
+        }
+    }
+
+    // should be overwritten by seperate shops 
+    public virtual void HandleItem(int weight) { 
+        GameObject pickedItem = GetRandomItem(weight);
+        if (spawnItem) {
+            Destroy(spawnedItem);
+            spawnedItem = Instantiate(pickedItem, transform.position, Quaternion.Euler(45, 0, 0));
+            spawnedItem.SetActive(true);
+        }
+    }
+
+    /*
```

Comment:
Use C# standard for commenting,
/// Rather than /* */
See https://doxygen.nl/manual/xmlcmds.html